compiler/rustc_borrowck/src/diagnostics/mod.rs

```diff
@@ -174,6 +186,10 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         }
     }
 
+    pub(super) fn describe_place_typed(&self, place_ref: PlaceRef<'tcx>) -> DescribedPlace {
```

Comment:
I presume this exists because changing `describe_place` to return a DescribedPlace` isn't practical?

Reply:
`describe_place` is used a lot, I'm currently narrowing the scope so it covers only `PlaceAndReason` cases.